### PR TITLE
Rename BaseColumn to DataFrameColumn

### PR DESCRIPTION
**Pull request by pgovind**
Rename BaseColumn to DataFrameColumn
Rename PrimitiveColumn to PrimitiveDataFrameColumn

I'm not renaming `ArrowStringColumn` -> `ArrowStringDataFrameColumn` because that's just too big IMO.

This patch is pure renaming, contains no new code. Should be an easy stamp.

### CODE REVIEW

File: src/Microsoft.Data/ArrowStringColumn.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ namespace Microsoft.Data
     /// <summary>
     /// An immutable to hold Arrow style strings
     /// </summary>
-    public partial class ArrowStringColumn : BaseColumn, IEnumerable<string>
+    public partial class ArrowStringColumn : DataFrameColumn, IEnumerable<string>
```
Comment (eerhardt) [inline src/Microsoft.Data/ArrowStringColumn.cs]: All other classes now have the `DataFrameColumn` suffix, but this one doesn't. Is there a reason for that?

File: src/Microsoft.Data/ArrowStringColumn.cs [hunk 2]
```diff
@@ -446,7 +446,7 @@ public override GroupBy GroupBy(int columnIndex, DataFrame parent)
             }
         }
 
-        public override BaseColumn FillNulls(object value, bool inPlace = false) => throw new NotSupportedException();
+        public override DataFrameColumn FillNulls(object value, bool inPlace = false) => throw new NotSupportedException();
```
Comment (eerhardt) [inline src/Microsoft.Data/ArrowStringColumn.cs]: (not related to this PR) But ArrowStringColumn should be able to `FillNulls` when `inPlace` is `false`, right? Maybe we should log an issue for that.

**Comment by eerhardt** [comment]
>                     return OrImplementation(column as PrimitiveDataFrameColumn<bool>, inPlace);

This is an existing issue, so you don't need to address it in this PR, but you don't need this extra cast here.

```C#
switch (column)
{
    case PrimitiveDataFrameColumn<bool> boolColumn:
        return OrImplementation(boolColumn, inPlace);
```

Will work correctly without the double type check.


---
Refers to: src/Microsoft.Data/PrimitiveDataFrameColumn.BinaryOperations.cs:249 in 8703944. [](commit_id = 870394406ed3d97a003ce7975d5e3230411a0842, deletion_comment = False)

File: src/Microsoft.Data/ArrowStringColumn.cs (hunk 2 above)
Comment (pgovind) [reply src/Microsoft.Data/ArrowStringColumn.cs]: Filed an issue for this

File: src/Microsoft.Data/ArrowStringColumn.cs (hunk 1 above)
Comment (pgovind) [reply src/Microsoft.Data/ArrowStringColumn.cs]: Renamed to `ArrowStringDataFrameColumn` now

**Pull request merged**